File: backend/E-commerce-Website/src/main/java/com/ecommercewebsite/service/CheckoutService.java
package com.ecommercewebsite.service;

import com.ecommercewebsite.dto.PaymentInfo;
import com.ecommercewebsite.dto.Purchase;
import com.ecommercewebsite.dto.PurchaseResponse;
import com.stripe.model.PaymentIntent;

public interface CheckoutService {

    PurchaseResponse placeOrder(Purchase purchase);
    PaymentIntent createPaymentIntent(PaymentInfo paymentInfo);
}

File: frontend/angular-ecommerce/src/app/services/checkout.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Purchase } from '../common/purchase';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { PaymentInfo } from '../common/payment-info';

@Injectable({
  providedIn: 'root'
})
export class CheckoutService {

  //update with https url using ENviroments
  private purchaseUrl = environment['luv2shopApiUrl'] + '/checkout/purchase';

  // private purchaseUrl = 'http://localhost:8080/api/checkout/purchase';

  //stripe payment url
  private paymentIntentUrl = environment['luv2shopApiUrl'] + '/checkout/payment-intent'; 

  constructor(private httpClient: HttpClient) { }

  placeOrder(purchase: Purchase): Observable<any>{

    return this.httpClient.post<Purchase>(this.purchaseUrl, purchase);
  }

//method for stripe payment Intent
  createPaymentIntent(paymentInfo: PaymentInfo): Observable<any> {
    return this.httpClient.post<PaymentInfo>(this.paymentIntentUrl,paymentInfo);
  }
}
